src/ledger/backend/readconn.ts

```diff
@@ -48,4 +56,25 @@ export class LedgerReadConn implements LedgerReader {
 	erdstall(): Address {
 		return Address.fromString(this.contract.address);
 	}
+
+	async getNftMetadata(token: Address, id: bigint): Promise<NFTMetadata> {
```

Comment:
If we keep metadata caching, we should give the user a way to make a fresh request, e.g. by adding to this function (and interface) an additional parameter `useCache?: bool`. I'm happy with keeping the default to true.

Reply:
Good point!